kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -118,19 +159,44 @@ mod tests {
         let domain = Radix2EvaluationDomain::new(n).unwrap();
         let rng = &mut StdRng::from_seed([0u8; 32]);
         let x = Fp::rand(rng);
-        let evaluator = LagrangeBasisEvaluations::new(domain, x);
+        let evaluator = LagrangeBasisEvaluations::new(domain.size(), domain, x);
 
         let expected = (0..n).map(|i| {
             let mut lagrange_i = vec![Fp::zero(); n];
             lagrange_i[i] = Fp::one();
-            Evaluations::from_vec_and_domain(lagrange_i, domain)
+            vec![Evaluations::from_vec_and_domain(lagrange_i, domain)
                 .interpolate()
-                .evaluate(&x)
+                .evaluate(&x)]
         });
 
-        for (i, expected) in expected.enumerate() {
-            if evaluator.evals[i] != expected {
-                panic!("{}, {}: {} != {}", line!(), i, evaluator.evals[i], expected);
+        for (i, (expected, got)) in expected.zip(evaluator.evals).enumerate() {
+            for (j, (expected, got)) in expected.iter().zip(got.iter()).enumerate() {
+                if got != expected {
+                    panic!("{}, {}, {}: {} != {}", line!(), i, j, got, expected);
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn test_new_with_chunked_segments() {
+        let n = 1 << 4;
+        let domain = Radix2EvaluationDomain::new(n).unwrap();
+        let rng = &mut StdRng::from_seed([0u8; 32]);
+        let x = Fp::rand(rng);
+        let evaluator = LagrangeBasisEvaluations::new(domain.size(), domain, x);
+        let evaluator_chunked =
+            LagrangeBasisEvaluations::new_with_chunked_segments(domain.size(), domain, x);
+        for (i, (evals, evals_chunked)) in evaluator
+            .evals
+            .iter()
+            .zip(evaluator_chunked.evals.iter())
+            .enumerate()
+        {
+            for (j, (evals, evals_chunked)) in evals.iter().zip(evals_chunked.iter()).enumerate() {
+                if evals != evals_chunked {
```

Comment:
So basically this test tries to check that the `if` clause in `new` indeed calls the chunked counterpart, no? Would it make sense to check the chunked evaluations against single-chunk evaluations from a larger domain?

Reply:
They will be different, unfortunately. The vector sum of the chunked evaluations should equal the unchunked ones, we could add that test if you think it's not adequately covered by https://github.com/o1-labs/proof-systems/pull/1033.